src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -149,31 +119,56 @@ const Instant& State::accessInstant() const
     return this->instant_;
 }
 
-const Position& State::accessPosition() const
+const Shared<const Frame> State::accessFrame() const
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("State");
+    }
+
+    return this->frameSPtr_;
+}
+
+const Position State::accessPosition() const
```

Comment:
This breaks convention a bit because we are now returning a copy of `Position` and `Velocity` rather than a reference to them. `accessPosition` is now functionally equivalent to `getPosition` (besides some errors checking).
`accessPosition` implies that we are able to modify the underlying value of position. This could be a breaking change if anything is trying to modifying `state->position_` using `accessPosition()`

Reply:
Yeah, `access` should return a reference, and `get` a copy (as a convention). Good catch!

Reply:
So the reason we removed the reference is because we have removed the member variables position_ and velocity_, therefore there is no reference, as there is no object. We can maintain the objects for backward compatibility but it will definitely not be very performant. But that's something we can improve in the future.

Reply:
The issue with that though is that we will have 2 coordinate arrays (one for state coordinates and one for position coordinates). If a user updates the position coordinates via accessPosition then they go out of sync... Gotta think about this a bit more.

Reply:
closing after agreeing on the topic, accessors removed.